hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -281,9 +277,9 @@ private void registerPartitionListener() {
         this.partitionLostRegistration =
                 getNodeEngine().getPartitionService().addPartitionLostListener(new PartitionLostListener() {
                     @Override
-                    public void partitionLost(PartitionLostEvent event) {
+                    public void partitionLost(final PartitionLostEvent event) {
                         // use toArray before iteration since it is done under mutex
-                        ScheduledFutureProxy[] futures = lossListeners.toArray(new ScheduledFutureProxy[lossListeners.size()]);
+                        ScheduledFutureProxy[] futures = lossListeners.toArray(new ScheduledFutureProxy[0]);
```

Comment:
Changed to 0, to let it get the size inside the mutex. Otherwise, the size might be smaller, inside the `toArray` causing the additional padding to be filled with nulls.